platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -375,28 +390,45 @@ private String getEntityId() {
     return String.format("https://%s:%s%s/saml", hostname, port, rootContext);
   }
 
-  private void logout() {
-    HttpSession session = sessionFactory.getOrCreateSession(request);
+  private SecurityAssertion getIdpSecurityAssertion() {
 
-    SecurityTokenHolder tokenHolder =
-        ((SecurityTokenHolder) session.getAttribute(SecurityConstants.SAML_ASSERTION));
+    return new SecurityAssertionImpl(getTokenHolder().getSecurityToken(IDP_REALM_NAME));
+  }
 
-    SecurityAssertion securityAssertion =
-        new SecurityAssertionImpl(tokenHolder.getSecurityToken("idp"));
+  private HttpSession getOrCreateSession() {
+    HttpSession session = sessionFactory.getOrCreateSession(request);
+    if (Objects.isNull(session)) {
+      throw new RuntimeException("Unable to get or create HTTP session");
+    }
+    return session;
+  }
 
-    boolean hasSecurityAuditRole =
-        Arrays.stream(System.getProperty("security.audit.roles").split(","))
-            .filter(role -> securityAssertion.getPrincipals().contains(new RolePrincipal(role)))
-            .findFirst()
-            .isPresent();
+  private void logout() {
+    logSecurityAuditRole();
+    getTokenHolder().remove(IDP_REALM_NAME);
+  }
 
-    if (hasSecurityAuditRole) {
+  private void logSecurityAuditRole() {
+    if (shouldAuditSubject()) {
       SecurityLogger.audit(
           "Subject with admin privileges has logged out: {}",
-          securityAssertion.getPrincipal().getName());
+          getIdpSecurityAssertion().getPrincipal().getName());
     }
+  }
 
-    tokenHolder.remove("idp");
+  private boolean shouldAuditSubject() {
+    return Arrays.stream(System.getProperty("security.audit.roles").split(","))
```

Comment:
:question: Is there any concern that the system property security.audit.roles can be maliciously modified to turn off auditing for roles?